### DISCUSSION TITLE
Differents QuickAccessToolBar in the same application

### DISCUSSION THREAD

**Discussion by InfodataJB** [fluentribbon/Fluent.Ribbon#1053] [Q&A] [answered]
Hello,
I am requesting your help about quickaccesstoolbar in fluent ribbon 6.0 (we can't upgrade from .net 4.5.1 at this time).
We have one application, that can connect into multi account. Each account as a different context and ribbon.
If I add the 2 first ribbon elements (2 buttons) in one account into the quick access toolbar, then we I logto another account (without ti change the app), I got the same elements as favorties in the quickaccess toolbar. But this elements are not related and in the other app I want to be able to configure the quickaccess
toolbar from scratch.
Could you please explain me if it is possible to configure and save one quickaccesstoolbar relative to each accounts/ribbons in the same .net application ? I don't know how tou store the quickaccesstoolbar on the local machine... May be I just need to apply a different ID to the ribbon (or to each elements of the ribbon).
Thank you for your help.

![Capture 1](https://user-images.githubusercontent.com/111052032/184117500-f0cc32ac-5d7b-4fed-bdb4-50a396fe234c.png)
![Capture 2](https://user-images.githubusercontent.com/111052032/184117509-2fa6a4df-3e45-4ead-995b-f7cca366101f.png)

**Comment by batzen** [comment, marked as answer]
The state storage is very simple and only uses index based lookup for items.
You can implement your own state storage by creating a class which implements `IRibbonStateStorage`.
To make use of it you have to create a new class inheriting from `Ribbon` and overwrite `CreateRibbonStateStorage` in that class.
If you want to load a different state after some action you can then simply call `ribbon.RibbonStateStorage.Load();`.
You can find the code for the default state storage at https://github.com/fluentribbon/Fluent.Ribbon/blob/v6.0.0/Fluent.Ribbon/Data/RibbonStateStorage.cs

**Reply by InfodataJB** [reply]
Thank you for your clear explanations!
